source/iofs/iofs.go

```diff
@@ -0,0 +1,36 @@
+// +build go1.16
+
+package iofs
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/golang-migrate/migrate/v4/source"
+	"github.com/golang-migrate/migrate/v4/source/httpfs"
+)
+
+func init() {
+	source.Register("iofs", &Iofs{})
+}
+
+// Iofs is a source driver for io/fs#FS.
+type Iofs struct {
+	httpfs.PartialDriver
```

Comment:
A refactor may be in-order.
It makes sense to go the other direction. e.g. add an `iofs.PartialDriver` and have `httpfs` embed that, since `io/fs.File` [interface](https://tip.golang.org/pkg/io/fs/#File) is a subset of the `net/http.File` [interface](https://golang.org/pkg/net/http/#File)